lib/mongo_client.js

```diff
@@ -217,31 +221,22 @@ var define = (MongoClient.define = new Define('MongoClient', MongoClient, false)
  * @deprecated MongoClient.connect is deprecated, please use new MongoClient().connect() to connect.
  */
 MongoClient.prototype.connect = function(callback) {
-  var self = this;
-
   // Validate options object
-  var err = validOptions(self.s.options);
-
-  // Return a promise
-  if (typeof callback !== 'function') {
-    return new this.s.promiseLibrary(function(resolve, reject) {
-      // Did we have a validation error
-      if (err) return reject(err);
-      // Attempt to connect
-      connect(self, self.s.url, self.s.options, function(err, topology) {
-        if (err) return reject(err);
-        self.topology = topology;
-        resolve(self);
-      });
-    });
+  var err = validOptions(this.s.options);
+
+  if (typeof callback === 'string') {
```

Comment:
why specifically check for a string?

Reply:
honestly, I didn't dig into it but left it there for legacy reasons. This was a big enough refactoring that it seemed important to reduce the variables in terms of code changes.

Reply:
oh oops, was this in another file before or something? (highlighted in green in the diff so I thought new)